website/site/content/docs/custom-previews.md

```diff
@@ -0,0 +1,216 @@
+edited. Netlify CMS uses [template compilers](#) to support different template languages. React
+components are supported without any compiler, and a Handlebars compiler is included by
+default.
+
+### Writing a Basic Template
+
+Let's look at an example template written in Handlebars. To create a template for use with a
+collection, we'll start by observing the `fields` configuration for the collection:
+
+```yaml
+# this snippet is from the main Netlify CMS configuration file, typically config.yml
+
+fields:
+  - { name: title, label: Title, widget: string }
+  - { name: text, label: Text, widget: text }
+```
+
+A preview template displaying these fields might look like this:
+
+```handlebars
+<div>
+  <h1>{{ title }}</h1>
+  <p>{{ text }}</p>
+</div>
+```
+
+You can add styles and classes as necessary to make the preview match your production site.
+
+
+### Registering Templates
+
+Once you have a template, you'll need to register it to be used with a collection. We're
+using a handlebars template, so we'll only need to provide a name for the template and the template
+itself.
+
+```js
+// You can pull in your template in a number of ways, but we'll go with an
+// inline template for simplicity.
+const template = `
+<div>
+  <h1>{{ title }}</h1>
+  <p>{{ text }}</p>
+</div>
+`
+
+CMS.registerPreviewTemplate('blog', template)
+```
+
+Registering a template may require other arguments depending on your template compiler. For more
+information on `registerPreviewTemplate`, check out the [API docs](#), as well as the docs for your
+template compiler.
+
+Now that you've registered your template, you'll want to add use it in one or more collections. This
+happens in your CMS configuration file via the `preview_template` key for a collection. Applying the
+"blog" template we just registered to a collection named "posts" would look like this:
+
+```yaml
+collections:
+  name: posts
+  label: Posts
+  preview_template: blog
+```
+
+### Using Field Previews
+
+When a preview template accesses a field value, it receives the raw value itself. This is the same
+value that is output when an entry is saved, and is produced by the field widget's [control
+component](#).  Many widgets also have a [preview component](#) that outputs an HTML representation of
+the field's value specifically for use in the preview pane. This is referred to as the Field
+Preview.
+
+We'll use a Markdown field to demonstrate. Here's a YAML file with a couple of fields:
+
+```yaml
+title: Blog Post
+description: A **brand new** post.
+```
+
+The "title" field is just a string value, but the "description" field contains markdown. We'll
+assume the configuration file is using the "markdown" widget for the description field.
+
+```handlebars
+<div>
+  <h1>{{ title }}</h1>
+  {{ description }}
+</div>
+```
+
+The above template accesses the value of the "description" field directly, so the preview pane would
+receive:
+
+```html
+<div>
+  <h1>Blog Post</h1>
+  A *brand new* post.
```

Comment:
The `description` above has two asterisks (`**brand new**`) but there's only one here (`*brand new*`).